kusto/ingest/result.go

```diff
@@ -0,0 +1,172 @@
+package ingest
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-kusto-go/kusto/ingest/internal/properties"
+	"github.com/Azure/azure-kusto-go/kusto/ingest/internal/resources"
+	"github.com/Azure/azure-kusto-go/kusto/ingest/internal/status"
+)
+
+// Result provides a way for users track the state of ingestion jobs.
+type Result struct {
+	record        statusRecord
+	tableClient   *status.TableClient
+	reportToTable bool
+	reportToQueue bool
+}
+
+// newResult creates an initial ingestion status record.
+func newResult() *Result {
+	ret := &Result{}
+
+	ret.record = newStatusRecord()
+	return ret
+}
+
+// putProps sets the record to a failure state and adds the error to the record details.
+func (r *Result) putProps(props properties.All) *Result {

```

Comment:
Here and below, I'm not sure why the methods are returning the type.  I looked at a couple places where you use putQueue() and putProps(), but I don't see you using that.  

Usually this is only done if you are going to do some type of chaining, like:

```go
  <-result.putQueue(mgr).putProps(props).Wait()
```

Reply:
I did this so that I can return res.PutProps() in a single line, but this is irrelevant now